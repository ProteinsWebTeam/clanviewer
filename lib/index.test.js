import ClanViewer from './index';
import * as d3 from 'd3-selection';

describe('constructor()', () => {
    it('should initialise the options', () => {
        const options = {
            "element":document.createElement('div'),
            "width":450,
            "height":250,
            "r":10,
            "testing":true
        };
        const obj = new ClanViewer(options);
        expect(obj).toMatchSnapshot();
    });
});

describe('#linkArc()', () => {
    it('should return the path for an arc', () => {
        const data = {
            "target":{"x":8, "y":2},
            "source":{"x":3, "y":6}
        };
        const arc = ClanViewer.linkArc(data);

        expect(arc).toMatchSnapshot();

    });
});

describe('#labelArc()', () => {
    it('should return the translate coordinates', () => {
        const data = {
            "target":{"x":8, "y":2},
            "source":{"x":3, "y":6}
        };
        ClanViewer.linkArc(data);
        const label = ClanViewer.labelArc(data);
        expect(label).toMatchSnapshot();
    });
});

describe('#processData(data)', () => {
    it('should create source and target objects in each element of data relationships', () => {
        // processData uses some values that should be initiated in the constructor, but given that we are calling
        // it in a static way, we set up those values here:
        const dataJson = {
            "members":[
                { "pfama_acc":"f1", "pfama_id":"id1", "num_occurrences":1},
                { "pfama_acc":"f2", "pfama_id":"id2", "num_occurrences":2}
            ],
            "relationships":[
                { "pfama_acc_1":"f1", "pfama_acc_2":"f2", "evalue":8.2e-6 }
            ]
        };
        const obj = new ClanViewer({r: 5});

        obj.processData(dataJson);
        expect(dataJson).toMatchSnapshot();
    });
});

describe('paint()', () => {
    const dataJson = {
        "members":[
            { "pfama_acc":"f1", "pfama_id":"id1", "num_occurrences":1},
            { "pfama_acc":"f2", "pfama_id":"id2", "num_occurrences":2}
        ],
        "relationships":[
            { "pfama_acc_1":"f1", "pfama_acc_2":"f2", "evalue":8.2e-6 },
            { "pfama_acc_1":"f1", "pfama_acc_2":"f2", "evalue":8.2e-7 },
            { "pfama_acc_1":"f2", "pfama_acc_2":"f1", "evalue":8.2e-5 }
        ]
    };
    it('should paint multiple relationships', function(){
        const root =document.createElement('div'),
            options = {
                "element": root,
                "testing":true,
                "multiple_relationships":true,
                "directional":true
            };
        const obj = new ClanViewer(options);

        obj.paint(dataJson);
        // eslint-disable-next-line no-unused-vars
        expect(obj.force.nodes().map(({x,y, ...rest}) => rest)).toMatchSnapshot();
        expect(dataJson.relationships).toHaveLength(d3.select(root).selectAll(".link").size());
        expect(dataJson.members).toHaveLength(d3.select(root).selectAll(".node").size());
        expect(d3.select(root).selectAll("marker").size()).toBe(1);
    });
});
