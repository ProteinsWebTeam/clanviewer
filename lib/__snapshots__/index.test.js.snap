// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#labelArc() should return the translate coordinates 1`] = `"translate(4.964101615137755,3.3301270189221936)"`;

exports[`#linkArc() should return the path for an arc 1`] = `"M3,6A6.4031242374328485,6.4031242374328485 0 0,1 8,2"`;

exports[`#processData(data) should create source and target objects in each element of data relationships 1`] = `
Object {
  "links": Array [
    Object {
      "score": 0.0000082,
      "source": "f1",
      "target": "f2",
    },
  ],
  "members": Array [
    Object {
      "num_occurrences": 1,
      "pfama_acc": "f1",
      "pfama_id": "id1",
    },
    Object {
      "num_occurrences": 2,
      "pfama_acc": "f2",
      "pfama_id": "id2",
    },
  ],
  "nodes": Array [
    Object {
      "accession": "f1",
      "name": "id1",
      "score": NaN,
    },
    Object {
      "accession": "f2",
      "name": "id2",
      "score": NaN,
    },
  ],
  "relationships": Array [
    Object {
      "evalue": 0.0000082,
      "pfama_acc_1": "f1",
      "pfama_acc_2": "f2",
    },
  ],
}
`;

exports[`constructor() should initialise the options 1`] = `
ClanViewer {
  "directional": false,
  "drag": [Function],
  "element": <div>
    <svg
      class="clanviewer"
      height="250"
      pointer-events="all"
      width="450"
    >
      <g />
    </svg>
  </div>,
  "force": Object {
    "alpha": [Function],
    "alphaDecay": [Function],
    "alphaMin": [Function],
    "alphaTarget": [Function],
    "find": [Function],
    "force": [Function],
    "nodes": [Function],
    "on": [Function],
    "restart": [Function],
    "stop": [Function],
    "tick": [Function],
    "velocityDecay": [Function],
  },
  "height": 250,
  "multiple_relationships": false,
  "r": 10,
  "size": [Function],
  "svg": Selection {
    "_groups": Array [
      Array [
        <g />,
      ],
    ],
    "_parents": Array [
      null,
    ],
  },
  "tickness": [Function],
  "width": 450,
  "zoom": [Function],
}
`;

exports[`paint() should paint multiple relationships 1`] = `
Array [
  Object {
    "accession": "f1",
    "index": 0,
    "name": "id1",
    "score": NaN,
    "vx": 0,
    "vy": 0,
  },
  Object {
    "accession": "f2",
    "index": 1,
    "name": "id2",
    "score": NaN,
    "vx": 0,
    "vy": 0,
  },
]
`;
